File: Algorithms/Stack implementation/stack.py
arr=[0,0,0,0,0]
maxi=5
top=-1
#inserting element at full stack
def push():
    global top,arr
    item=input("Enter element to push: ")
    if top==maxi-1:
        print("Overflow")
    else:
        top=top+1
        arr[top]=item
        print("Array after pushing ",item ," is:",arr)

#deleting element in empty stack
def pop():
    global top,arr
    if top==-1:
        print("underflow")
    else:
        print("Array element popped is:",arr[top])
        arr[top]=0
        top=top-1
        print(arr)

def peep():
    global top
    print(top)

def main():
    i=1
    while True:
        print("--------STACK OPERATIONS-----------")
        print("1.Push")
        print("2.Pop")
        print("3.Peep")
        print("4.Exit")
        print("-----------------------")
        choice=int(input("Enter your choice:"))
        if choice==1:
            push()
        elif choice==2:
            pop()
        elif choice==3:
            peep()
        elif choice==4:
            exit(0)
        else:
            print("Invalid choice")
main()

File: linked list/ds/stack.c
#include<stdio.h>

int isempty();
int isfull();
int peak();
void push();
int pop();
void display();

int top=-1,i,go=-1;
char s[100],ma[100];int main()      
{
	int ch;
	    
	    do       
        {
	    printf("\n\n---- STACK----\n\tMenu ");
	    printf(" \n1.push\n2.pop\n3.peak\n4.Display\n");
	    printf("Enter your choice(1-5):");
	    scanf("%d",&ch);
	    switch(ch)
	    { 
            case 1:push();
                    break;
	        case 2:pop();
                    break;
	        case 3: peak();
                    break;
	        case 4: display();
	                break;
	        default: printf("Wrong Choice!!");
	    }
	    
    }while (ch<=5);
	    return 0;
}
    

int isempty()
    {
        if (go==-1)
        return 1;
        else
        return 0;
        
    }
int isfull(){
        if(go==99)
        return 1;
        else 
        return 0;
    }
void push()
    {
        if(isfull())
        printf("Stack is overflow\n");
        else
        {
            char ch;
            printf("Enter the element");
            scanf("%ch",&ch);
            if (ch=='+'){
                top++;
                printf("man");
                s[top]=ch;}
            if (ch=='-'){
                for(i=top;i>=-1;i--){
                    if (s[i]!='+')
                    {
                        go++;
                        ma[go]=s[i];
                    }
                    top=top-i;
                    s[top++]=ch;
                    }
                }
            if (ch=='*'){
                for(i=top;i>=-1;i--){
                    if (s[i]!='+'|| s[i]!='-')
                    {
                        go++;
                        ma[go]=s[i];
                    }
                    top=top-i;
                    s[top]=ch;
                    }
                }
                if (ch=='%'){
                for(i=top;i>=-1;i--){
                    if (s[i]!='+'||s[i]!='-'||s[i]!='*')
                    {
                        go++;
                        ma[go]=s[i];
                    }
                }
                    top=top-i;
                    s[top]=ch;
                    
                if (ch=='('){
                    s[top]=ch;
                    }
                if (ch==')'){
                    for(i=top;i>=-1;i--){
                        if (s[i]!='(')
                        {
                            go++;
                            ma[go]=s[i];
                        }
                    }
                        top=top-i;
                    }
                else {
                    
                        go++;
                        ma[go]=s[i];
                }
                }
            }
        }
    }

int pop(){
        if(isempty())
        printf("stack is underflow\n");
        else
        {
            return s[top--];
        }
    }

int peak(){
        if(isempty())
        printf("stack is underflow\n");
        else
        {
            return s[top];
        }
    }
void display()
    {
        if(isempty())
        printf("stack is empty");
        else
        {
            for(i=go;i>=0;i--){
                printf("%d\t",ma[i]);     
            }
        }
    }

